### PR TITLE
Changed to debug log

### PR DESCRIPTION
**Pull request by htahir1**
## Describe changes
I wanted to clean up log output

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by schustmi — COMMENTED** [review]
I don't like this change tbh, we should reduce the amount but this is something security relevant which users should not do

**Comment by htahir1** [comment]
@schustmi I understand but this should only be given once and it shouldnt be in the constructor.. Otherwise its spammy as hell. I think we can raise it when people are registering a component

**Comment by schustmi** [comment]
Registering seems like a good compromise to me!

**Comment by schustmi** [comment]
Maybe we can add a parameter like `warn_about_plain_text_secrets` to `StackComponentConfig.__init__` which defaults to `False`, and then in `Client.create_stack_component(...)` we explicitly set it to `True`? This would only show the warning when registering a component, but the warning will not be shown in the dashboard (which should probably handle it differently anyway, e.g. using a dynamic warning message or automatically creating a secret)

File: src/zenml/config/secret_reference_mixin.py [hunk 1]
```diff
@@ -51,8 +54,11 @@ def __init__(self, **kwargs: Any) -> None:
                 continue
 
             if not secret_utils.is_secret_reference(value):
-                if secret_utils.is_secret_field(field):
-                    logger.warning(
+                if (
+                    secret_utils.is_secret_field(field)
+                    and warn_about_plain_text_secrets
+                ):
+                    logger.debug(
```
Comment (schustmi) [inline src/zenml/config/secret_reference_mixin.py:61]:
```suggestion
                    logger.warning(
```

**Review by schustmi — CHANGES_REQUESTED** [review]

**Review by schustmi — APPROVED** [review]

**Pull request merged**